### ISSUE TITLE
Rendering Rmd in Atom

### ISSUE THREAD

**Issue by AtomicNess123** [fangzhou-xie/atomknitr#1]
I read your article: https://fangzhou-xie.github.io/posts/2020-10-28-atomrmd/
Are you not using this method to compile Rmd? 
I did not manage to make it work. Does this need language-r package or Rbox would suffice?

**Comment by fangzhou-xie** [comment]
Thanks for you interest!

You should refer the blog post for compiling RMD files on your system, instead of this package. The result PDF should be in the same directory as your RMD file. This is because `rmarkdown::render('{FileName}')` is the function that actually handles the compilation process, and the default argument controlling for the output directory is the same as your RMD file. Hence they will be in same folder.

Now, regarding why you shouldn't use this package, this is because I used a bit of hack to make it work for me to compile the RMD docs for my website. And this will most certainly **NOT** work for you, unless you have an identical setup as I do. If you read the source code at [line 23](https://github.com/fangzhou-xie/atomknitr/blob/master/R/atomknitr.R#L23), the output directory is `../_posts` (because I put RMD docs for my website in a different folder than the result HTML files). Moreover, there are several other lines that adds my local Zotero database for reference, and also removes automatically generated HTML headers etc. 

So basically this function is a sequence of hacks that I used to make it work, **only** for building the HTML files for the blogs. If you need to build your RMD files to PDFs, you can just refer to the blog post; if you need more than that (like what I did), then you should understand how to customize the `rmarkdown::render()` function and maybe write a package of yourself to use.

**Comment by AtomicNess123** [comment]
Thanks! Yes, I understand that your package is highly personalized for your use.
What I tried to say is that I followed your steps in your blog to make Rmd compile in Atom, but after I restart (just in case) I press Ctrl + 2, and nothing goes on. Is there something else I should do? I have installed all packages, corrected the cson file, and opened a Rmd file.

**Comment by fangzhou-xie** [comment]
Several questions: 
1. Are you compiling to PDF or HTML? 
2. Do you see the output file (for example, `NameOfYourFile.html`) in the same folder of your RMD file after hitting Ctrl+2? 
3. When hitting Ctrl+2, you should be able to have a console message like this.
<img width="516" alt="Screen Shot 2021-07-30 at 9 24 42 AM" src="https://user-images.githubusercontent.com/45321577/127659604-e2f5ffd0-7de3-47a5-9947-e66716ea3932.png">
When you say "nothing goes on", do you mean you can't see this message at all? Or is there an error message for this?

**Comment by AtomicNess123** [comment]
Thanks for your patience :)

I have on my config.cson:

```
"atom-shell-commands":
  commands: [
    {
      arguments: [
        "-e"
        "rmarkdown::render('{FileName}')"
      ]
      command: "R"
      name: "knitRmd"
      options:
        cwd: "{FileDir}"
        keymap: "ctrl-2"
        save: true
    }
  ]

```

Now, when I check my keybindings, I don't see Ctrl+2:

![image](https://user-images.githubusercontent.com/47029233/127662115-57e1fc39-4d6f-4136-862e-79c2f53562b7.png)

So that is a problem to start with. When I try to access the menu, I only see this:

![image](https://user-images.githubusercontent.com/47029233/127662307-f2b467c3-6f75-4d08-9116-02f2df5c7748.png)

So I have no way to run this.

Oh, and I am compiling to both PDF and HTML:

```
output:
  html_document:
    runtime: shiny
    df_print: paged
    toc: yes
  pdf_document:
    highlight: tango
    toc: yes
```

**Comment by fangzhou-xie** [comment]
I think it is because of the `config.cson` file. 

Try this:

```
"*":
  "atom-shell-commands":
  commands: [
    {
      arguments: [
        "-e"
        "rmarkdown::render('{FileName}')"
      ]
      command: "R"
      name: "knitRmd"
      options:
        cwd: "{FileDir}"
        keymap: "ctrl-2"
        save: true
    }
  ]
```

I think the `"*"` part is to apply this configuration globally. Let me know if this works for you and I might need to update my blog post as well.

**Comment by AtomicNess123** [comment]
It was the indenting! The * was on top (I did not copy pasted all my config file).
One last question: is it possible to preview the RMarkdown as PDF or html side by side?
Thanks for your help!

**Comment by fangzhou-xie** [comment]
Great! Glad that you could make it work.

If by "side by side" you mean "showing the result of the PDF while writing RMD in real-time", then unfortunately the answer is no. Not to my knowledge at least. There are packages that will render HTML output from Markdown in real-time, but I haven't found a way to do that for RMD. Maybe the best way of writing RMD is really inside RStudio :)

**Comment by AtomicNess123** [comment]
Thanks for your answer. Is in Rstusio possible? You mean the pop-up window that shows the rendered document?

**Comment by fangzhou-xie** [comment]
No. From my experience, even RStudio cannot render the output in real-time. As you have observed, there will be a pop-up window for the rendered document, and if you don't close that window, the next time you "knit" the RMD, the output doc will be updated faster than closing the window and rendering it again. But I still think RStudio has the support for RMD better.

I guess the real difficulty is the lengthy compilation of LaTeX. I haven't heard any editor that could render PDF in real-time (except the LaTeX official editor, was it called TexShop?). When compiling RMD to PDF, the compiler will first compile RMD into plain MD, and then to PDF. This process is very slow even for very simple documents and I assume any effort trying to make this process real-time will be futile.

**Comment by AtomicNess123** [comment]
Thanks, all clear now :)

**Comment by fangzhou-xie** [comment]
If you are using RStudio 1.4, you may find this useful (https://blog.rstudio.com/2020/09/30/rstudio-v1-4-preview-visual-markdown-editing/). I don't think this is exactly what you want (real-time side-by-side rendering), but it does offer the output rendered in real time.

**Comment by AtomicNess123** [comment]
Thanks a lot, will have a look for sure!
